flink-connector-aws/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/table/RowDataKinesisDeserializationSchema.java

```diff
@@ -118,6 +118,11 @@ public RowData deserialize(
             throws IOException {
 
         RowData physicalRow = physicalDeserializer.deserialize(recordValue);
+        // If message can not be deserialized by physicalDeserializer - return null to skip record
+        if (physicalRow == null) {
+            return null;
+        }
```

Comment:
Assume this only happens if the json deserializer is configured to ignore?

Reply:
Yes, this is part of `DeserializationSchema` interface behaviour:
> @return The deserialized message as an object (null if the message cannot be deserialized).